unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/TextShape/ECSTextShapeComponentHandler.cs

```diff
@@ -32,25 +33,36 @@ public ECSTextShapeComponentHandler(AssetPromiseKeeper_Font fontPromiseKeeper, I
         this.renderersInternalComponent = renderersInternalComponent;
     }
 
+    private void OnTextRendererUpdated(TMP_TextInfo tmproInfo)
+    {
+        renderersInternalComponent.RemoveFor(scene, entity);
+        renderersInternalComponent.AddRenderer(scene, entity, textRenderer);
+    }
+
     public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
     {
+        this.scene = scene;
+        this.entity = entity;
+
         textGameObject = new GameObject("TextShape");
 
-        textRenderer = textGameObject.AddComponent<MeshRenderer>();
         rectTransform = textGameObject.AddComponent<RectTransform>();
         textComponent = textGameObject.AddComponent<TextMeshPro>();
+        textRenderer = textComponent.renderer;
         rectTransform.SetParent(entity.gameObject.transform, false);
         rectTransform.localRotation = Quaternion.Euler(0, 180, 0);
 
         textComponent.text = string.Empty;
         textComponent.richText = true;
         textComponent.overflowMode = TextOverflowModes.Overflow;
-        renderersInternalComponent.AddRenderer(scene, entity, textRenderer);
+        textComponent.OnPreRenderText += OnTextRendererUpdated;
     }
 
     public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
     {
-        renderersInternalComponent.RemoveRenderer(scene, entity, textRenderer);
+        textComponent.OnPreRenderText -= OnTextRendererUpdated;
+        renderersInternalComponent.RemoveFor(scene, entity);
```

Comment:
we can't remove renderersInternalComponent for this entity, since an entity might have other renderers attached to it (gltf, meshrenderer, etc)